### DISCUSSION TITLE
Question: How do you handle dynamic routes?

### DISCUSSION THREAD

**Discussion by approached** [vinissimus/next-translate#410] [Q&A] [answered]
Hi @ all

How do you handle dynamic routes?
Example:
```
EN: /news
DE: /neuigkeiten
RU: /новостI

**Comment by aralroca** [comment, marked as answer]
@approached All the i18n routing part is from Next.js core, so maybe this can help you: https://github.com/vercel/next.js/discussions/18885

**Comment by approached** [comment]
@aralroca thank you for the link. A really great solution doesn't exists now.
The routing function from nextjs currently ugly https://nextjs.org/docs/api-reference/next.config.js/rewrites#rewrites-with-i18n-support

**Reply by TomasSestak** [reply]
Could you please let me know if you figure out solution?

**Reply by approached** [reply]
You must create a routing wrapper and add rewrites https://nextjs.org/docs/api-reference/next.config.js/rewrites#rewrites-with-i18n-support 

I know it's a ugly solution.

**Reply by TomasSestak** [reply]
could you please include your example code?
